source/guide/3.2/guide-security.md

```diff
@@ -11,33 +11,230 @@ pageord: 500
 
 # Overview
 
-This guide will quickly explain how to bootstrap a secured manager and use if from the cli and web UI.
+This guide will explain how to bootstrap a secured manager and use it from the CLI and web UI. <br/>
+Securing the manager focuses on the REST service, which is the access point of all clients to the management server.
+
+Cloudify's security framework uses the [Flask-secuREST](https://github.com/cloudify-cosmo/flask-securest/tree/0.6)
+package, which integrates with [Flask-RESTful](https://flask-restful.readthedocs.org/en/0.3.2/) to secure REST services.
+
+When security is enabled, all requests to the REST service are authenticated before they can reach their endpoint.
+For example, when a user of the Web-UI attempts to see the all blueprints, a request is sent to the REST service
+"blueprints" endpoint, but it will only reach it if the user is logged in with valid credentials. <br/>
+Similarly, the CLI command "cfy deployments create" will only be successful if valid
+username and password are sent with the REST call to the endpoint "deployments".
+Of course, direct requests made using other clients (e.g. curl) must also include correct credentials, or they will
+fail with an "Unauthorized User" error.
+
 
 # Main Concepts
 
-## Userstores
+## Userstore
+Generally, a userstore is simply a class that enables loading of user details and returns them as a user object.
+
+Typically (but not always) user details are stored as records in a database or objects in a directory. Each user can
+be identified by a unique attribute, such as a username or id, or by a unique combination of attributes.<br/>
+In order to authenticate a user (for example by a set of username and password) it might be required to load the user's
+details and verify the given credentials indeed match.<br/>
+To support a variety of user-store systems and configurations Cloudify security framework can accept different
+userstore implementations. It's possible to use the default Flask-secuREST simple userstore or to specify a new
+implementation that supports a specific userstore system.
+
+A valid userstore implementation can be any Python class that inherits from
+[abstract_userstore.py](https://github.com/cloudify-cosmo/flask-securest/blob/master/flask_securest/userstores/
+abstract_userstore.py) and implements a `get_user` method.
+The object returned by `get_user` must adhere to Flask-secuREST's User Model, specifying these 3 methods:
+
+- is_active()
```

Comment:
probably requires more information on these here.. or a link to someplace else which has it